### PR TITLE
[ENH] Modernization

### PR DESCRIPTION
**Pull request by rksh**
working through legacy cruft and resource management problems that became apparently when upgrading to SDK 31

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/ThemeUtil.java [hunk 1]
```diff
@@ -16,12 +18,24 @@
...
-            AppCompatDelegate.setDefaultNightMode(displayMode);
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            Handler handler = new Handler(Looper.getMainLooper());
+            executor.execute(() -> {
+                final int displayMode = prefs.getInt(ListFragment.PREF_DAYNIGHT_MODE, AppCompatDelegate.MODE_NIGHT_YES);
+                Logging.info("set theme called: " + displayMode);
+                handler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        AppCompatDelegate.setDefaultNightMode(displayMode);
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ui/ThemeUtil.java:35]: Does this really take long enough to need a background thread?

**Review by bobzilladev — APPROVED** [review]
Seems good! A lot of things going to executors, I think the DB helper will lock it properly, but some UI setting things as well which I'm surprised to see.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 2]
```diff
@@ -2705,31 +2635,31 @@ public void backgroundUploadFile(){
...
-            AlertDialog.Builder iseDlgBuilder = new AlertDialog.Builder(this);
-            iseDlgBuilder.setMessage(external?R.string.no_external_space_message:R.string.no_internal_space_message)
-                    .setTitle(external?R.string.no_external_space_title:R.string.no_internal_space_title)
-                    .setCancelable(true)
-            .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
-                @Override
-                public void onClick(DialogInterface dialog, int which) {
-                    dialog.dismiss();
+    public void checkStorage() {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java:2639]: we should consider (for future work) if we want to have a single executor for these uses.

**Review by uhtu — COMMENTED** [review]
👍

**Pull request merged**